Add reporting document for package metadata

mattgarrish opened the pull request:
This is a palceholder document to use to report on authoring of the various package metadata.

One modification I've made is to ensure these will only ever be two-column tables. The dpub model used a column for each publisher because we knew before making the report document only three publishers were needed to cover the full range of values. Given there will be different uptake of different properties here, and we'll probably need to talk to a lot more people to get sufficient coverage of each property, using the column approach could get ugly to read.

In place of that approach, I've labelled the second column "used by" and added a placeholder list to each row where we can identify who is using each property. I've done the same to the a11y report for consistency.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/epub-specs/pull/2002.html" title="Last updated on Feb 25, 2022, 5:06 PM UTC (fa3eee9)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/epub-specs/2002/5ba7886...fa3eee9.html" title="Last updated on Feb 25, 2022, 5:06 PM UTC (fa3eee9)">Diff</a>

mattgarrish committed:
add reporting document for package metadata;
update a11y report to use the same two-column table structure

iherman commented on epub33/reports/a11y-properties-use.md:
```diff
@@ -66,25 +86,41 @@ implementations.
     </thead>
     <tbody>
         <tr>
         	<td>dcterms:conformsTo</td>
```
As a matter of consistency, shouldn't these be a `<th>` and not a `<td>`? It is the case everywhere else (including the other document).

mattgarrish replied:
Each row is a property, so this seems unnecessary.

If you're referring to the element/attribute/property definitions in the core spec, each row is a new piece of information. That's why the first cell in each row is a th rather than there being a th that covers the whole column.

iherman replied:
No, I am referring to the report. If you look at, e.g., `schema:accessibilitySummary`, the term itself is in bold, whereas, say, `dcterms:conformsTo` is not. All the similar terms in the other document are also in bold.

mattgarrish replied:
Oh, I see what you mean. The other document isn't correct. I did a global replace to format the rows and looks like I incorrectly input th for the property names. Going to undo that now.

iherman commented on epub33/reports/epub-properties-use.md:
```diff
@@ -0,0 +1,604 @@
+
+The following table lists publishers who have stated that they are currently using
+the meta properties in production.
+
```
This is a general comment, both for this section and the other document: it may be better to put active links to the specification themselves. It helps the reviewers to find their way through the report.

Also, we should discuss whether to set up the two-way links between the specifications and this document the same way as we have done for the functional tests (recognizing that, if we do the latter, it may be better to convert these documents into HTML, which is of course doable but it is a drag). Again, this may help the reviewers to see the completeness (or not...) of our "tests".

Cc @dlazin

mattgarrish replied:
> we should discuss whether to set up the two-way links

It probably wouldn't be too hard to do this. I assume the script for the test links checks if specStatus=ED before generating?

iherman replied:
This is done by respec, not by my script; my stuff just generates the reports. 

However. I realize now that it straight out of the box to use the respec trick. Respec uses the `testSuiteURI`: it sees a `data-test` value (e.g., `#pgk=lang_pkg`), concatenates that with the value of `testSuiteURI` and then creates the small pull-down values (and yes, it does it only for the editors' draft). I do not think it is prepared to use several `testSuiteURI` values.

That being said, there is already a `data-test-display.js` script that massages those boxes a bit. It would be possible to find out, somehow, that those links are to be used with a different URI...

But we still have to have the right target fragment. Adding a fragment to a markdown my be a pain.

mattgarrish replied:
I've linked all the properties to their definitions in the latest commit. I've also added links to any other references (not just to the specs but also tools).

For completeness, I've also added a paragraph to each section saying where the properties are used. Might help with understanding what the properties do. Worst case it doesn't harm anything.

mattgarrish replied:
I haven't tried to link from the specification to the report.

iherman commented on epub33/reports/epub-properties-use.md:
```diff
@@ -0,0 +1,604 @@
+        </tr>
+        <tr>
+            <th>meta-auth (deprecated)</th>
+            <td>
```
I would think we could leave this one out. There is no specification for it in our document; it is redirected to EPUB 3.0 after all...

iherman commented on epub33/reports/epub-properties-use.md:
```diff
@@ -0,0 +1,604 @@
+            </td>
+        </tr>
+        <tr>
+            <th>marc21xml-record (deprecated)</th>
```
Same comments as before (and valid for the two entries below and to `xml-signature` and `xmp-record`): I wonder whether listing deprecated properties here makes sense.

This is also a question to @dlazin @wareid @dauwhe

iherman replied:
Thinking a bit further, back in the specification proper: maybe regrouping the deprecated records separately in, eg, §D3, may make the situation cleaner...

mattgarrish replied:
I put them in for completeness. What I was thinking we would do is add some sort of boilerplate statement in the second column about deprecated properties not needing verification. I can try and figure something out now, but I wouldn't shift them around into a new group. That'll just make it harder to locate them. We should try to parallel the core spec as much as possible.

mattgarrish replied:
What I've done for these is strike them through using `s` tags and dropped "(deprecated)" and instead added the prose:

> This property is deprecated and no longer recommended for use in EPUB Publications. It is only listed for completeness of reporting.

mattgarrish committed:
add gca partners to a11y report

mattgarrish committed:
add linking;
use s tag on deprecated properties;
add ids for properties missing them in specs;
fix misuse of th on property names

mattgarrish committed:
fix unrecognized md inside table cells

mattgarrish committed:
remove lists of gca partners
